### PR TITLE
added   "--no-sandbox" option for chrome webdriver

### PR DESCRIPTION
**Pull request by reckoner**
Otherwise, the Chrome webdriver will fail on Windows 10 WSL2 Ubuntu18 and on containerized environments.

All pull requests must have an associated issue in the issue tracker. If there
isn't one, please go open an issue describing the defect, deficiency or desired
feature. You can read more about our issue and PR processes in the
[wiki](https://github.com/bokeh/bokeh/wiki/BEP-1:-Issues-and-PRs-management).

- [X] issues: fixes ##10461
- [x] tests added / passed
- [x] release document entry (if new feature or API change)

### CODE REVIEW

**Comment by bryevdv** [comment]
AFAIK Google itself recommends against setting this option, so adding this would need be be behind a specific feature flag (that defaults to off) or else part of a general configuration mechanism that lets users pass in any configuration to the driver.

**Comment by mattpap** [comment]
> (...) else configurable via a general mechanism that lets users pass in any options to the driver.

This would be my preferred approach, as it would allow to configure a driver more precisely when needed.

**Comment by mattpap** [comment]
`--no-sandbox` is a very unsafe feature and should be only used on case-by-case basis, definitively not as a default in `bokeh.io`.

**Comment by bryevdv** [comment]
@reckoner Will you be able to continue working on the PR to make things configurable? I think the ideal outcome is something like
```
create_chromium_webdriver(extra_options=["--foo", "--bar"])
```

and then just loop over `extra_options` called `options.add_option` on each. 

I would also suggest `extra_options` be kwarg-only and have a default value of `None`

File: bokeh/io/webdriver.py [hunk 1]
```diff
@@ -87,13 +87,16 @@ def create_firefox_webdriver() -> WebDriver:
         service_log_path=devnull,
     )
 
-def create_chromium_webdriver() -> WebDriver:
+def create_chromium_webdriver(extra_options=None) -> WebDriver:
```
Comment (bryevdv) [inline bokeh/io/webdriver.py:90]: @reckoner 
```suggestion
def create_chromium_webdriver(extra_options: Optional[List[str]] = None) -> WebDriver:
```

Comment (bryevdv) [reply bokeh/io/webdriver.py:90]: `Iterable[str]` probably best but that will also need a new import above

File: bokeh/io/webdriver.py [hunk 2]
```diff
@@ -87,13 +87,16 @@ def create_firefox_webdriver() -> WebDriver:
...
     options = webdriver.chrome.options.Options()
     options.add_argument("--headless")
     options.add_argument("--hide-scrollbars")
     options.add_argument("--force-device-scale-factor=1")
     options.add_argument("--force-color-profile=srgb")
-    options.add_argument("--no-sandbox")
+    if extra_options:
+      for op in extra_options:
+          assert isinstance(op,str)
```
Comment (bryevdv) [inline bokeh/io/webdriver.py:98]: This assertion can be dropped, let `options.add_argument` adjudicate bad params

**Comment by bryevdv** [comment]
@reckoner thanks for the update, a few small comments, also our linter will complain if indentation is not 4 spaces:
```
E         bokeh/io/webdriver.py:97:7: E111 indentation is not a multiple of four
E         bokeh/io/webdriver.py:98:11: E111 indentation is not a multiple of four
E         bokeh/io/webdriver.py:99:11: E111 indentation is not a multiple of four
```

**Comment by bryevdv** [comment]
@reckoner Thank you for the changes, this PR is nearly there. Will you be able to add the MyPy type as described above?

**Comment by bryevdv** [comment]
@reckoner did you intend to close the PR? It had been accepted and was going to be merged.

**Comment by reckoner** [comment]
Sorry. I accidentally closed this.

**Comment by bryevdv** [comment]
@reckoner apologies, somehow our default branch was not set correctly (it was set to `main` but should have been set to `banch-2.3` for the next release cycle. I had hoped just changing here in the UI would work, but it really needs a proper rebase on `branch-2.3`Can you either rebase+force push or else close this and reopen a new PR with a branch off of `branch-2.3`? Thank you for your patience

**Comment by bryevdv** [comment]
@reckoner will you be able to rebase this off `branch-2.3`? If not I will close this PR and leave the issue as a `GOOD FIRST ISSUE` Sorry for the churn we are adopting a new branching policy and there are some growing pains.

**Comment by reckoner** [comment]
I created a new pull request. 

https://github.com/bokeh/bokeh/pull/10489

**Pull request closed without merging**